### PR TITLE
SCRUM-1403 Autopopulate 'Modified By' field on editing

### PR DESCRIPTION
**Pull request by markquintontulloch**
Field populated with first and last name

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/services/helpers/validators/DiseaseAnnotationValidator.java [hunk 1]
```diff
@@ -198,13 +192,11 @@ public DiseaseAnnotation validateCommonDiseaseAnnotationFields(DiseaseAnnotation
...
         if (createdBy != null) dbEntity.setCreatedBy(createdBy);
 
-        String modifiedBy = validateModifiedBy(uiEntity);
-        if (modifiedBy != null) dbEntity.setModifiedBy(modifiedBy);
-    
         if (uiEntity.getCreationDate() != null)
             dbEntity.setCreationDate(uiEntity.getCreationDate());
         
         dbEntity.setDateLastModified(OffsetDateTime.now());
+        dbEntity.setModifiedBy(authenticatedPerson.getFirstName() + " " + authenticatedPerson.getLastName());
```
Comment (oblodgett) [inline src/main/java/org/alliancegenome/curation_api/services/helpers/validators/DiseaseAnnotationValidator.java:199]: @markquintontulloch im not sure this is the best place to set this... I would suggest not setting, these in the validator but rather in the service.

Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/helpers/validators/DiseaseAnnotationValidator.java:199]: @oblodgett I'd agree, except that unless we do some refactoring we would then need this code in the GeneDiseaseAnnotationService, AlleleDiseaseAnnotationService, and AGMDiseaseAnnotationService as that's where the update method gets called from

Comment (oblodgett) [reply src/main/java/org/alliancegenome/curation_api/services/helpers/validators/DiseaseAnnotationValidator.java:199]: Yes I agree... we need this at a much upper level. Also basically advocating for these four fields should be at the base of every class. I would suggest have a base service method that does this and that would work for ALL objects.

Comment (markquintontulloch) [reply src/main/java/org/alliancegenome/curation_api/services/helpers/validators/DiseaseAnnotationValidator.java:199]: Feels like that would be part of a larger ticket to refactor code if/when we change the AuditedObject model

**Review by oblodgett — APPROVED** [review]

**Pull request merged**